sourcecred/sourcecred
# Pagination for explorer home

# Description
This PR adds pagination, filtering, and sorting for the Activity Table on Explorer Home via the TableState util. Sorting can be done via cred and grain, and is done by clicking on their header labels on the table. Filtering can be done by name, entering text into the box at the top right, or by type of participant (participant, organization, bot, project) with the checkboxes at the bottom. Also, the Explorer Home page now uses the proper currency name and suffix, loaded from config (defaults to "Grain" and "g", respectively).

Closes issue #2464 

# Test Plan
Tested manually with the following workflows
- Load page and confirm that default sort is by cred, descending
- Click on cred header and confirm that sort is now by cred, ascending
- Click on grain header and confirm that sort is by grain, ascending (assumed since grain is 0 for all users in data set)
- Click on grain header again and confirm that sort is by grain, descending (assumed since grain is 0 for all users in data set)
- Choose different pagination options and confirm that the proper number of rows are shown
- Click forward and back on the pages, and confirm that the proper page of results is shown
- Type letters into the name filter at top and confirm that only names that match filter string are shown (same letters, in order, but not necessarily sequential)
- Confirm that averages and totals for cred and grain are correct for visible list entries filtered by name
- Check various checkboxes for participant type at bottom of list, and confirm that only participants of said type are visible
- Confirm that averages and totals for cred and grain are correct for visible entries filtered by type
